test/spec/services/apiServiceSpec.js

```diff
@@ -234,6 +235,47 @@ describe("APIService", function(){
       });
     }));
   });
-  */
+
+
+  describe('#availableKinds', function() {
+    var bothSample = ['Binding','ConfigMap','DeploymentConfig','Event','LimitRange','Pod','ReplicaSet','Role','Service', 'Template'];
+    var onlyClusterSample = ['ClusterResourceQuota','Namespace','OAuthAccessToken','PersistentVolume','ProjectRequest','User'];
+
+    // at minimum, prove with tests that it is deduplicating by ignoring openshift!
+    it('should return list of kinds that are scoped to a namespace by default', function() {
+      var namespacedKinds = _.map(APIService.availableKinds(), 'kind');
+      // from the fixtures included, we should have 51 namespaced kinds
+      expect(namespacedKinds.length).toEqual(51);
+      expect( _.difference(bothSample, namespacedKinds).length ).toEqual(0);
+    });
+
+    it('should return list of all kinds, including those that are cluster scoped, when passed a truthy argument', function() {
+      var allKinds = _.map(APIService.availableKinds(true), 'kind');
+      // from the fixtures included, we should have 84 total kinds
+      expect(allKinds.length).toEqual(84);
+      expect( _.difference(bothSample, allKinds).length ).toEqual(0);
+      expect( _.difference(onlyClusterSample, allKinds).length ).toEqual(0);
+    });
+
+    it('should not duplicate kinds that exist in both an api group & the openshift namespace', function() {
+      var allKinds =  _.map(APIService.availableKinds(true), 'kind');
+      var counts = _.reduce(allKinds, function(result, kind) {
+        if(!result[kind]) {
+          result[kind] = 0;
+        }
+        result[kind]++;
+        return result;
+      }, {});
+      // TODO: need a review, not sure why these are still duplicated
```

Comment:
@spadgett @jwforres need a comment here.  These two still end up with duplication, not sure why. Assuming it is ok?